### PR TITLE
Babel v7, ESlint fixes, React/React Navigation upgrades, mobx removed

### PR DESCRIPTION
**Pull request by daviscabral**
This PR is just to follow the changes I am doing - while airbnb/javascript repo is not fully on Babel v7 - I had to build it locally (I really hate monorepos for situations like this).

### CODE REVIEW

**Comment by daviscabral** [comment]
@aksonov that's because I am using a local copy of `eslint-config-airbnb`. I'll push one temporary fix for it - while they do not release it.

**Comment by aksonov** [comment]
I guess scenes.test.js should be disabled now because it works with internal state...

**Comment by daviscabral** [comment]
Cool - only linter errors now.

**Comment by aksonov** [comment]
What about removing dist and use src? react-navigation doesn't do compilation now, right?

**Comment by daviscabral** [comment]
Done - using src now :+1:.

File: src/navigationStore.js [hunk 1]
```diff
@@ -692,25 +704,26 @@ class NavigationStore {
...
-    } else if (tabs) {
-      if (!tabBarComponent) {
-        tabBarComponent = tabBarPosition === 'top' ? props => <TabBarTop {...props} {...commonProps} /> : props => <TabBarBottom {...props} {...commonProps} />;
-      }
-      if (!tabBarPosition) {
-        tabBarPosition = Platform.OS === 'android' ? 'top' : 'bottom';
+    }
+
+    if (tabs) {
+      let createTabNavigator = createMaterialTopTabNavigator;
```
Comment (daviscabral) [inline src/navigationStore.js]: `TabBarTop` and `TabBarBottom` are deprecated.

**Comment by aksonov** [comment]
Yes, we need to replace it them

**Comment by daviscabral** [comment]
I did already - but that limited a bit the way it can be customized.

**Comment by aksonov** [comment]
Yes, just found it. Don't worry, we will think about it later.

**Comment by aksonov** [comment]
Let me know when I can merge

**Comment by daviscabral** [comment]
Feel free to merge it - I got a few other things to keep going and I'll need to postpone the rest of changes for later tonight or tomorrow morning.

**Comment by aksonov** [comment]
No rush, will merge tomorrow then

**Comment by daviscabral** [comment]
@aksonov initial work done.

**Comment by aksonov** [comment]
Well done, merging!

**Pull request merged**